src/extension/src/utils/dependencyChecker.ts

```diff
@@ -0,0 +1,80 @@
+import { WizardServant, IPayloadResponse } from "../wizardServant";
+import { ExtensionCommand, CONSTANTS } from "../constants";
+const os = require('os');
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+const PYTHON3_REGEX = RegExp('^Python 3\\.[5-9]\\.[0-9]');       // minimum Python version required is 3.5.x
+const NODE_REGEX = RegExp('v10\\.(1[5-9]|[2-9][0-9])\\.[0-9]');  // minimum Node version required is 10.15.x
+
+export class DependencyChecker extends WizardServant {
+  clientCommandMap: Map<
+    ExtensionCommand,
+    (message: any) => Promise<IPayloadResponse>
+  >;
+
+  constructor() {
+    super();
+    this.clientCommandMap = this.defineCommandMap();
+  }
+
+  private defineCommandMap(): Map<
+    ExtensionCommand,
+    (message: any) => Promise<IPayloadResponse>
+  > {
+    return new Map([[ExtensionCommand.CheckDependency, this.checkDependency]]);
+  }
+
+  private async runPythonVersionCommand(command: string) {
+    let installed: boolean;
+    try {
+      const { stdout } = await exec(command + ' --version');
+      if (PYTHON3_REGEX.test(stdout)) {
+        installed = true;
+      } else {
+        installed = false;
+      }
```

Comment:
NIT: could be simplified to `installed = PYTHON3_REGEX.test(stdout)`

Reply:
done